raiden-ts/src/helpers.ts

```diff
@@ -53,16 +54,47 @@ export const getContracts = (network: Network): ContractsInfo => {
   }
 };
 
+/**
+ * Generate, sign and return a subkey from provided main account
+ *
+ * @param network - Network to include in message
+ * @param main - Main signer to derive subkey from
+ * @returns Subkey's signer & address
+ */
+async function genSubkey(network: Network, main: Signer) {
+  const url = globalThis?.location?.origin ?? 'unknown';
+  const message = `=== RAIDEN SUBKEY GENERATION ===
+
+Network: ${getNetworkName(network).toUpperCase()}
+Raiden dApp URL: ${url}
+
+WARNING: ensure this signature is being requested from Raiden dApp running at URL above by comparing it to your browser's url bar.
+Signing this message at any other address WILL give it FULL control of this subkey's funds, tokens and Raiden channels!`;
+
+  const signature = await main.signMessage(toUtf8Bytes(message));
+  const pk = sha256(signature) as Hash;
+  const signer = new Wallet(pk, main.provider);
+
+  return { signer, address: signer.address as Address };
+}
+
 /**
  * Returns a [[Signer]] based on the `account` and `provider`.
  * Throws an exception if the `account` is not a valid address or private key.
  *
  * @param account - an account used for signing
  * @param provider - a provider
+ * @param subkey - Whether to generate a subkey
  * @returns a [[Signer]] or [[Wallet]] that can be used for signing
  */
-export const getSigner = async (account: string | number | Signer, provider: JsonRpcProvider) => {
+export const getSigner = async (
+  account: string | number | Signer,
+  provider: JsonRpcProvider,
+  subkey?: boolean,
```

Comment:
`subKey`

Reply:
No, `subkey` was the keyword used everywhere, it's not two words, but a single one representing the `subkey` flag.